src/components/clusterexplorer/explorer.ts

```diff
@@ -120,24 +124,51 @@ export class KubernetesExplorer implements vscode.TreeDataProvider<ClusterExplor
             if (ti.collapsibleState === vscode.TreeItemCollapsibleState.None && this.extenders.some((e) => e.contributesChildren(element))) {
                 ti.collapsibleState = vscode.TreeItemCollapsibleState.Collapsed;
             }
+            if (!element.customizedTreeItem) {
+                this.customizePossibly(element, ti);
+            }
         });
 
-        let customisedTreeItem = extensionAwareTreeItem;
+        return extensionAwareTreeItem;
+    }
+
+    async customizePossibly(element: ClusterExplorerNodeV2, extensionAwareTreeItem: vscode.TreeItem) {
+        const customizedTreeItem = Object.assign({}, extensionAwareTreeItem);
         for (const c of this.customisers) {
-            customisedTreeItem = providerResult.transformPossiblyAsync(extensionAwareTreeItem, (ti) => c.customize(element, ti));
+            await c.customize(element, customizedTreeItem);
+        }
+        if (!this.shallowEqual(customizedTreeItem, extensionAwareTreeItem)) {
```

Comment:
Mmmm not sure, this will fail if the treeItem and the customizedTreeItem have different properties keys and values. So, theoretically it should work. BTW if an extender customize a treeItem this should always have the `customizedTreeItem` property (which doesn't belong to the original treeItem), no? So they should always be different somehow